psiturk/psiturk_shell.py

```diff
@@ -941,60 +944,89 @@ def hit_create(self, numWorkers, reward, duration):
+                                 '  template size uploadable to the Ad server is',
+                                 '  1048576 bytes!' % size_of_ad])
+                return
 
-        if self.tunnel.is_open:
-            ip_address = self.tunnel.url
-            port = str(self.tunnel.tunnel_port)  # Set by tunnel server.
-        else:
-            ip_address = str(self.web_services.get_my_ip())
-            port = str(self.config.get('Server Parameters', 'port'))
-        ad_content = {
-            'psiturk_external': True,
-            'server': ip_address,
-            'port': port,
-            'browser_exclude_rule': str(self.config.get('HIT Configuration', 'browser_exclude_rule')),
-            'is_sandbox': int(self.sandbox),
-            'ad_html': ad_html,
-            # 'amt_hit_id': hitid, Don't know this yet
-            'organization_name': str(self.config.get('HIT Configuration', 'organization_name')),
-            'experiment_name': str(self.config.get('HIT Configuration', 'title')),
-            'contact_email_on_error': str(self.config.get('HIT Configuration', 'contact_email_on_error')),
-            'ad_group': str(self.config.get('HIT Configuration', 'ad_group')),
-            'keywords': str(self.config.get('HIT Configuration', 'psiturk_keywords'))
-        }
+            # what all do we need to send to server?
+            # 1. server
+            # 2. port
+            # 3. support_ie?
+            # 4. ad.html template
+            # 5. contact_email in case an error happens
 
-        create_failed = False
-        fail_msg = None
-        ad_id = self.web_services.create_ad(ad_content)
-        if ad_id is not False:
+            if self.tunnel.is_open:
+                ip_address = self.tunnel.url
+                port = str(self.tunnel.tunnel_port)  # Set by tunnel server.
+            else:
+                ip_address = str(self.web_services.get_my_ip())
+                port = str(self.config.get('Server Parameters', 'port'))
+            ad_content = {
+                'psiturk_external': True,
+                'server': ip_address,
+                'port': port,
+                'browser_exclude_rule': str(self.config.get('HIT Configuration', 'browser_exclude_rule')),
+                'is_sandbox': int(self.sandbox),
+                'ad_html': ad_html,
+                # 'amt_hit_id': hitid, Don't know this yet
+                'organization_name': str(self.config.get('HIT Configuration', 'organization_name')),
+                'experiment_name': str(self.config.get('HIT Configuration', 'title')),
+                'contact_email_on_error': str(self.config.get('HIT Configuration', 'contact_email_on_error')),
+                'ad_group': str(self.config.get('HIT Configuration', 'ad_group')),
+                'keywords': str(self.config.get('HIT Configuration', 'psiturk_keywords'))
+            }
 
+            create_failed = False
```

Comment:
Seems like there is a fair amount of code duplication here, especially with `hit_config`... it looks to me like the major differences are the `ad_location` and then the logic afterwards that checks if it failed or not, so I think it should be possible to at least share the code for the `hit_config` (maybe make a function to generate the hit config, and allow passing in the `ad_location` manually?)